### PR TITLE
fix: Remove allowances support from airdrop transaction

### PR DESCRIPTION
**Pull request by ibankov**
**Description**:

Remove allowances support from airdrop transaction, and add tests for it.

**Related issue(s)**:

Fixes #

**Notes for reviewer**:

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoTransferValidator.java [hunk 1]
```diff
@@ -190,11 +192,16 @@ public static void validateTokenTransfers(List<TokenTransferList> tokenTransfers
...
-            final List<AccountAmount> fungibleTransfers, final Set<Pair<AccountID, Boolean>> uniqueTokenAcctIds)
+            final List<AccountAmount> fungibleTransfers,
+            final Set<Pair<AccountID, Boolean>> uniqueTokenAcctIds,
+            final boolean isAirdrop)
             throws PreCheckException {
         validateTruePreCheck(isNetZeroAdjustment(fungibleTransfers), TRANSFERS_NOT_ZERO_SUM_FOR_TOKEN);
         boolean nonZeroFungibleValueFound = false;
         for (final AccountAmount acctAmount : fungibleTransfers) {
+            if (isAirdrop) {
+                validateFalsePreCheck(acctAmount.isApproval(), INVALID_TRANSACTION);
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/validators/CryptoTransferValidator.java:203]: We could throw NOT_SUPPORTED here or specific response here ?

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM except response code

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java [hunk 2]
```diff
@@ -306,7 +306,7 @@ final var record = createPendingAirdropRecord(pendingId, null);
      * the credits for the receivers.
      * @param fungibleAmounts the fungible airdrop amounts
      * @param sender the sender account id
-     * @param isApproval if the airdrop transfer is sent with an approval
+     * @param isApproval should be false for airdrop
```
Comment (Neeharika-Sompalli) [inline hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java:309]: probably just add enum `AllowanceStrategy` 
 with `ALLOWANCES_ALLOWED` and `ALLOWANCES_REJECTED` for it to be readable ?

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Just one comment. Can be done in a different PR.

File: hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java (hunk 2 above)
Comment (ibankov) [reply hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/TokenAirdropHandler.java:309]: Will do in different PR.

**Pull request merged**